services/storage/src/simcore_service_storage/modules/celery/_celery_types.py

```diff
@@ -0,0 +1,50 @@
+from functools import partial
+from pathlib import Path
+from typing import Any
+
+from kombu.utils.json import register_type  # type: ignore[import-untyped]
+from pydantic import BaseModel
+
+
+def _path_encoder(obj):
+    if isinstance(obj, Path):
+        return {"__path__": True, "path": str(obj)}
+    return obj
+
+
+# Define how Path objects are deserialized
+def _path_decoder(obj):
+    if "__path__" in obj:
+        return Path(obj["path"])
+    return obj
+
+
+def _class_full_name(clz: type[BaseModel]) -> str:
+    return ".".join([clz.__module__, clz.__qualname__])
+
+
+def _encoder(obj: BaseModel, *args, **kwargs) -> dict[str, Any]:
+    return obj.model_dump(*args, **kwargs)
+
+
+def _decoder(clz: type[BaseModel], data: dict[str, Any]) -> BaseModel:
+    return clz(**data)
+
+
+def _register_pydantic_types(*models: type[BaseModel]) -> None:
```

Comment:
Q: is this not used? I see it's private

Reply:
actually Celery will be compatible with Pydantic in version 5.5 which is the next minor version

Reply:
will remove that stuff